File: Algorithms/1013.Partition-Array-Into-Three-Parts-With-Equal-Sum/partition_array_into_three_parts_with_equal_sum.py
class Solution:
    def canThreePartsEqualSum(self, A: List[int]) -> bool:
        total = sum(A)
        if total % 3 != 0:
            return False
        
        equal = total // 3
        part = 0
        count = 0
        i = 0
        j = 0
        for idx in range(0, len(A)):
            part += A[idx]
            if part == equal:
                part = 0
                count += 1
                if count == 1 :
                    i = idx
                else:
                    j = idx + 1
                
        if count == 3 and i + 1 < j:
            return True
        return False
            

File: Algorithms/1013.Partition-Array-Into-Three-Parts-With-Equal-Sum/partition_array_into_three_parts_with_equal_sum.go
func canThreePartsEqualSum(A []int) bool {
    total := 0
    for _, value := range A {
        total += value
    }
    
    if total % 3 != 0 {
        return false
    }
    count := 0
    part := total / 3
    cur := 0
    for _, value := range A {
        cur += value
        if cur == part {
            count ++
            cur = 0
        }
    }
    if count == 3 {
        return true
    }
    
    
    return false
}